mapper/pojo/src/main/java/org/hibernate/search/mapper/pojo/mapping/impl/PojoTypeWorkExecutor.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.mapper.pojo.mapping.impl;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Supplier;
+
+import org.hibernate.search.engine.backend.document.DocumentElement;
+import org.hibernate.search.engine.backend.index.spi.DocumentReferenceProvider;
+import org.hibernate.search.engine.backend.index.spi.IndexWorkExecutor;
+import org.hibernate.search.mapper.pojo.session.context.spi.PojoSessionContextImplementor;
+
+public class PojoTypeWorkExecutor<I, E, D extends DocumentElement> {
```

Comment:
Good call. I don't think we discussed it, but adding a intermediary "type executor" here can definitely help.